analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/SolutionBuilder.cs

```diff
@@ -73,28 +73,23 @@ internal struct SolutionBuilder
                 throw new ArgumentException("Please use a collection of paths with the same extension", nameof(paths));
             }
 
-            var project = Create()
+            return Create()
                 .AddProject(AnalyzerLanguage.FromPath(paths.First()), outputKind: outputKind)
                 .AddDocuments(paths)
-                .AddReferences(additionalReferences);
-
-            return project.GetSolution();
+                .AddReferences(additionalReferences)
+                .Solution;
         }
 
         public static SolutionBuilder FromSolution(Solution solution) =>
-            new SolutionBuilder(solution);
+            new(solution);
 
-        public IReadOnlyList<Compilation> Compile(params ParseOptions[] parseOptions)
+        public ImmutableArray<Compilation> Compile(params ParseOptions[] parseOptions) =>
+            solution.Projects.SelectMany(x => Compile(x, parseOptions)).ToImmutableArray();
+
+        private static IEnumerable<Compilation> Compile(Project project, ParseOptions[] parseOptions)
         {
-            var options = ParseOptionsHelper.GetParseOptionsOrDefault(parseOptions);
-
-            return Solution
-                .Projects
-                .SelectMany(project => options
-                    .Where(ParseOptionsHelper.GetFilterByLanguage(project.Language))
```

Comment:
This line was hiding an issue when a VB project was compiled against CSharpParseOptions.